XiEditor/EditViewController.swift

```diff
@@ -107,7 +112,7 @@ class EditViewController: NSViewController, EditViewDataSource {
         let first = Int(floor(scrollView.contentView.bounds.origin.y / textMetrics.linespace))
         let height = Int(ceil((scrollView.contentView.bounds.size.height) / textMetrics.linespace))
         let last = first + height
-        if first != firstLine || last != lastLine && document != nil {
```

Comment:
Yea, my reasoning here is just to avoid silent failures. If this is getting called too early then we're doing setup at the wrong point in the view controller lifecycle, and that's the thing to change. There is no reason for us to ever send an event before the document is set.